### PR TITLE
feat(Merriam-Webster): add presence

### PR DESCRIPTION
**Pull request by unicornbetrayal**
## Description 

Added a presence for [Merriam-Webster](https://www.merriam-webster.com/)
Resolves https://github.com/PreMiD/Presences/issues/5480

## Acknowledgements
- [x] I read the [Presence Guidelines](https://github.com/PreMiD/Presences/blob/main/.github/CONTRIBUTING.md)
- [x] I linted the code by running `yarn format`
- [x] The PR title follows the repo's [commit conventions](https://github.com/PreMiD/Presences/blob/main/.github/COMMIT_CONVENTION.md)

## Screenshots

Attached are a few possible scenarios, there are many more.

<details>

image
image
image
image
image

<summary> Proof showing the creation/modification is working as expected </summary>

</details>

### CODE REVIEW

File: websites/M/Merriam-Webster/presence.ts [hunk 1]
```diff
@@ -0,0 +1,271 @@
+const presence = new Presence({
+		clientId: "1002734097719894067",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/12lTZIs.png",
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:8]: Go to https://discord.com/developers/applications => the application for this presence => rich presence => upload the logo with e.g. the name "logo"
```suggestion
			largeImageKey: "logo",
```

File: websites/M/Merriam-Webster/presence.ts [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = window.location;
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:11]:
```suggestion
		{ pathname, href } = window.location;
```

File: websites/M/Merriam-Webster/presence.ts [hunk 3]
```diff
@@ -0,0 +1,267 @@
...
+			.querySelector("head > title")
+			.textContent.replace(" Definition & Meaning - Merriam-Webster", "")
+			.replace(
+				" - Definition and More from the Free Merriam-Webster Dictionary",
+				""
+			)}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:27]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+presence.on("UpdateData", async () => {
+	const presenceData: PresenceData = {
+			largeImageKey: "https://i.imgur.com/12lTZIs.png",
+			startTimestamp: browsingTimestamp,
+		},
+		{ pathname } = window.location;
+
+	if (document.location.pathname === "/")
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:13]:
```suggestion
	if (pathname === "/")
```

File: websites/M/Merriam-Webster/presence.ts [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+		presenceData.details = "Browsing the dictionary";
+	else if (document.location.pathname.startsWith("/dictionary/")) {
+		presenceData.details = "Viewing definitions";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" Definition & Meaning - Merriam-Webster", "")
+			.replace(
+				" - Definition and More from the Free Merriam-Webster Dictionary",
+				""
+			)}`;
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:23]: Look for better selectors that don't require all that replacing.
```suggestion
		presenceData.state = document.querySelector("#left-content > div:nth-child(2) > div:nth-child(1) > h1").textContent
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Merriam-Webster/presence.ts [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Merriam-Webster Games And Quizzes", "")
+			.replace(" | Merriam-Webster Games & Quizzes", "")
+			.replace(" - Word Game | Merriam-Webster", "")
+			.replace(" | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "Play",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:41]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 7]
```diff
@@ -0,0 +1,267 @@
...
+	} else if (document.location.pathname.startsWith("/word-of-the-day")) {
+		presenceData.details = "Viewing the word of the day";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace("Word of the Day: ", "")
+			.replace(" | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:53]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 8]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/words-at-play/")) {
+		presenceData.details = "Reading post";
+		presenceData.state = `${document
+			.querySelector("meta[property='og:title']")
+			.getAttribute("content")}`;
+		presenceData.buttons = [
+			{
+				label: "View Post",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:64]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 9]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/reviews/")) {
+		presenceData.details = "Reading a review";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Reviews by Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Review",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:75]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 10]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/video/")) {
+		presenceData.details = "Watching a video";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" (Video) | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "Watch Video",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:86]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 11]
```diff
@@ -0,0 +1,267 @@
...
+			.getAttribute("content")
+			.replace("Thesaurus results for ", "");
+		presenceData.details = "Viewing synonyms";
+		presenceData.state = `${
+			word.charAt(0).toUpperCase() + word.slice(1).toLowerCase()
+		}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:101]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 12]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/help/")) {
+		presenceData.details = "Viewing the help page";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:112]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 13]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/vocabulary/")) {
+		presenceData.details = "Viewing vocabulary learning lists";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:123]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 14]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/browse/legal/")) {
+		presenceData.details = "Browsing the law dictionary";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(": Browse the Dictionary | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:134]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 15]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/browse/medical/")) {
+		presenceData.details = "Browsing the medical dictionary";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(": Browse the Dictionary | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:145]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 16]
```diff
@@ -0,0 +1,267 @@
...
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(
+				" Definition & Meaning | Merriam-Webster Legal",
+				""
+			)}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:159]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 17]
```diff
@@ -0,0 +1,267 @@
...
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(
+				" Definition & Meaning | Merriam-Webster Medical",
+				""
+			)}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:173]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 18]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/about-us/")) {
+		presenceData.details = "Viewing the about page";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:184]:
```suggestion
				url: href,
```

File: websites/M/Merriam-Webster/presence.ts [hunk 19]
```diff
@@ -0,0 +1,267 @@
...
+		];
+	} else if (document.location.pathname.startsWith("/contact-us/")) {
+		presenceData.details = "Viewing the contact page";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Merriam-Webster", "")}`;
+		presenceData.buttons = [
+			{
+				label: "View Page",
+				url: location.href,
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:195]:
```suggestion
				url: href,
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

File: websites/M/Merriam-Webster/presence.ts [hunk 20]
```diff
@@ -0,0 +1,267 @@
...
+				label: "Play",
+				url: location.href,
+			},
+		];
+	} else if (document.location.pathname.startsWith("/word-of-the-day")) {
+		presenceData.details = "Viewing the word of the day";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace("Word of the Day: ", "")
+			.replace(" | Merriam-Webster", "")}`;
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:49]:
```suggestion
		presenceData.state = document.querySelector('[class="word-and-pronunciation"]').firstElementChild.textContent
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Comment by unicornbetrayal** [comment]
Thank you for the suggestions @DarkVillager

File: websites/M/Merriam-Webster/presence.ts (hunk 1 above)
Comment (EncryptedDev) [reply websites/M/Merriam-Webster/presence.ts]: No, including the Imgur link is preferable for future maintainability so the author doesn't have to be contacted to change assets.

Comment (EncryptedDev) [reply websites/M/Merriam-Webster/presence.ts]: @unicornbetrayal Please revert this

Comment (unicornbetrayal) [reply websites/M/Merriam-Webster/presence.ts]: > @unicornbetrayal Please revert this

Okay, done

Comment (DarkVillager) [reply websites/M/Merriam-Webster/presence.ts:8]: > No, including the Imgur link is preferable for future maintainability so the author doesn't have to be contacted to change assets.

Good to know. Will implement this when I get the chance.

File: websites/M/Merriam-Webster/presence.ts [hunk 21]
```diff
@@ -0,0 +1,264 @@
...
+			{
+				label: "View Page",
+				url: href,
+			},
+		];
+	} else if (document.location.pathname.startsWith("/words-at-play/")) {
+		presenceData.details = "Reading post";
+		presenceData.state = `${document
+			.querySelector("meta[property='og:title']")
+			.getAttribute("content")}`;
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:57]: Same as above
```suggestion
		presenceData.state = document.querySelector('[class="title-lg"]').textContent
```

File: websites/M/Merriam-Webster/presence.ts [hunk 22]
```diff
@@ -0,0 +1,264 @@
...
+			{
+				label: "View Post",
+				url: href,
+			},
+		];
+	} else if (document.location.pathname.startsWith("/reviews/")) {
+		presenceData.details = "Reading a review";
+		presenceData.state = `${document
+			.querySelector("head > title")
+			.textContent.replace(" | Reviews by Merriam-Webster", "")}`;
```
Comment (DarkVillager) [inline websites/M/Merriam-Webster/presence.ts:68]: Same as above
```suggestion
		presenceData.state = document.querySelector('[class="text-center"]').textContent
```

**Review by DarkVillager — CHANGES_REQUESTED** [review]

**Review by EncryptedDev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**